### PR TITLE
feat(draw_sw): extend lv_draw_sw_rotate with L8 support

### LINKED ISSUE #6518: Software rotating (180°) of LV_COLOR_FORMAT_L8 doesn't work
### LVGL version

v9.1.0

### What happened?

I use the SSD1322 display controller and the color format L8. I tried to rotate the display by 180° but nothing happens. There's no specific rotate function implemented for L8.

### How to reproduce?

lv_draw_sw_rotate(src, dst, width, height, width, width, LV_DISPLAY_ROTATION_180, LV_COLOR_FORMAT_L8);

### PR DESCRIPTION
**Pull request by kisvegabor**
fixes #6518

### Description of the feature or fix

A clear and concise description of what the bug or new feature is.

### Notes
- Update the [Documentation](https://github.com/lvgl/lvgl/tree/master/docs) if needed.
- Add [Examples](https://github.com/lvgl/lvgl/tree/master/examples) if relevant.
- Add [Tests](https://github.com/lvgl/lvgl/blob/master/tests/README.md) if applicable.
- If you added new options to `lv_conf_template.h` run [lv_conf_internal_gen.py](https://github.com/lvgl/lvgl/blob/master/scripts/lv_conf_internal_gen.py) and update [Kconfig](https://github.com/lvgl/lvgl/blob/master/Kconfig).
- Run `scripts/code-format.py` ([astyle](http://astyle.sourceforge.net/install.html) version [v3.4.12](https://github.com/szepeviktor/astyle/releases/tag/v3.4.12) needs to be installed) and follow the [Code Conventions](https://docs.lvgl.io/master/CODING_STYLE.html).
- Mark the Pull request as [Draft](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/changing-the-stage-of-a-pull-request) while you are working on the first version, and mark is as _Ready_ when it's ready for review.
- When changes were requested, [re-request review](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/requesting-a-pull-request-review) to notify the maintainers.
- Help us to review this Pull Request! Anyone can [approve or request changes](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/reviewing-changes-in-pull-requests/approving-a-pull-request-with-required-reviews).

### CODE REVIEW

**Review by liamHowatt — DISMISSED** [review]
LGTM. The choice to use camel case surprised me though.

**Review by liamHowatt — DISMISSED** [review]
Thanks, sorry for the hassle

**Review by FASTSHIFT — DISMISSED** [review]

**Comment by kisvegabor** [comment]
Fixed merge conflict. Please review again.

**Review by liamHowatt — DISMISSED** [review]

**Comment by kisvegabor** [comment]
Code formatting... Please approve it again. Sorry.

**Review by liamHowatt — APPROVED** [review]

**Comment by kisvegabor** [comment]
@FASTSHIFT @XuNeo Please review this PR.

**Review by XuNeo — APPROVED** [review]

File: src/draw/sw/lv_draw_sw.c [hunk 1]
```diff
@@ -550,23 +589,23 @@ static void execute_drawing(lv_draw_sw_unit_t * u)
...
+                               int32_t dst_stride)
 {
-    if(LV_RESULT_OK == LV_DRAW_SW_ROTATE90_ARGB8888(src, dst, srcWidth, srcHeight, srcStride, dstStride)) {
+    if(LV_RESULT_OK == LV_DRAW_SW_ROTATE90_ARGB8888(src, dst, src_width, src_height, src_stride, dst_stride)) {
         return ;
     }
 
-    srcStride /= sizeof(uint32_t);
-    dstStride /= sizeof(uint32_t);
+    src_stride /= sizeof(uint32_t);
```
Comment (XuNeo) [inline src/draw/sw/lv_draw_sw.c:600]: Please check if this is true. The name suggests it's already stride

Comment (kisvegabor) [reply src/draw/sw/lv_draw_sw.c:600]: The input stride is in bytes but we convert it to pixels, which is correct as the loops below also work on pixels.

**Pull request merged**